src/test/scala/org/labrad/ServerTest.scala

```diff
@@ -234,6 +235,19 @@ class ServerTest extends FunSuite with AsyncAssertions {
     assert(keys.contains("a"))
     assert(!keys2.contains("a"))
   }
+
+  test("server gets context expiration message when client disconnects") { fix =>
+    val ctx = withClient(fix.host, fix.port, fix.password) { client2 =>
+      val ctx = client2.newContext.copy(high = client2.id)
+      await(client2.send("Scala Test Server", ctx, "Set" -> Cluster(Str("a"), Str("test"))))
+      ctx
+    }
+    Thread.sleep(1000)
```

Comment:
This is usually a warning sign in java tests that you're going to have flaky (and thus possibly next to useless) tests. 

The ways to get around this are to inject clocks, use a fake threading framework, fake task scheduling framework, etc so that you programatically determing which tasks get run.